ratpack-dropwizard-metrics/src/main/java/ratpack/dropwizard/metrics/internal/HttpClientMetrics.java

```diff
@@ -0,0 +1,114 @@
+import ratpack.dropwizard.metrics.HttpClientConfig;
+import ratpack.exec.ExecController;
+import ratpack.http.client.HttpClient;
+import ratpack.http.client.HttpClientStats;
+import ratpack.service.Service;
+import ratpack.service.StartEvent;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+@Singleton
+public class HttpClientMetrics implements Service, Runnable {
+
+  private static final String METRIC_PREFIX = "httpclient.";
+  private static final String TOTAL_ACTIVE_CONNECTIONS = getMetricName("total.active.connections");
+  private static final String TOTAL_IDLE_CONNECTIONS = getMetricName("total.idle.connections");
+  private static final String TOTAL_CONNECTIONS = getMetricName("total.connections");
+
+  private final HttpClient httpClient;
+  private final MetricRegistry metricRegistry;
+  private final HttpClientConfig config;
+  private final Map<String, HttpMetricGauge> gauges;
+
+  @Inject
+  public HttpClientMetrics(
+    HttpClient httpClient,
+    MetricRegistry metricRegistry,
+    DropwizardMetricsConfig config
+  ) {
+    this.httpClient = httpClient;
+    this.metricRegistry = metricRegistry;
+    this.config = config.getHttpClient();
+    this.gauges = new HashMap<>();
+  }
+
+  @Override
+  public void onStart(StartEvent event) throws Exception {
+    if (config.isEnabled()) {
+      ExecController execController = event.getRegistry().get(ExecController.class);
+      execController.getExecutor().scheduleAtFixedRate(this, 0, config.getPollingFrequencyInSeconds(), TimeUnit.SECONDS);
+    }
+  }
+
+  @Override
+  public void run() {
+    HttpClientStats httpClientStats = httpClient.getHttpClientStats();
+    gauge(TOTAL_ACTIVE_CONNECTIONS).setValue(httpClientStats.getTotalActiveConnectionCount());
+    gauge(TOTAL_IDLE_CONNECTIONS).setValue(httpClientStats.getTotalIdleConnectionCount());
+    gauge(TOTAL_CONNECTIONS).setValue(httpClientStats.getTotalConnectionCount());
+    httpClientStats.getStatsPerHost().forEach((host, stats) -> {
+        gauge(getHostMetricName(host, "total.active.connections"))
+          .setValue(stats.getActiveConnectionCount());
+        gauge(getHostMetricName(host, "total.idle.connections"))
+          .setValue(stats.getIdleConnectionCount());
+      gauge(getHostMetricName(host, "total.connections"))
+          .setValue(stats.getTotalConnectionCount());
+      });
+  }
+
+  private HttpMetricGauge gauge(String name) {
+    if (gauges.containsKey(name)) {
+      return gauges.get(name);
+    }
+    HttpMetricGauge gauge = new HttpMetricGauge();
+    gauges.put(name, gauge);
+    return metricRegistry.register(name, gauge);
+  }
+
+  private static String getMetricName(String name) {
+    return METRIC_PREFIX + name;
+  }
+
+  private static String getHostMetricName(String host, String name) {
+    return getMetricName(host + "." + name);
+  }
+
+  private static class HttpMetricGauge implements Gauge<Long> {
```

Comment:
Also felt a bit weird making my own gauge.  Couldn't find an implementation that let me just set the value.  Ideas here welcome.